src/algebra/tropical/big_operators.lean

```diff
@@ -0,0 +1,160 @@
+No concrete (semi)ring is used here, only ones with inferrable order/lattice structure, to support
+real, rat, ereal, and others (erat is not yet defined).
+
+Minima over `list α` are defined as producing a value in `with_top α` so proofs about lists do not
+directly transfer to minima over multisets or finsets.
+
+-/
+
+
+open_locale big_operators
+
+variables {R S : Type*}
+
+open tropical
+
+lemma list.trop_sum [add_monoid R] (l : list R) : trop l.sum = list.prod (l.map trop) :=
+begin
+  induction l with hd tl IH,
+  { simp },
+  { simp [←IH] }
+end
+
+lemma multiset.trop_sum [add_comm_monoid R] (s : multiset R) :
+  trop s.sum = multiset.prod (s.map trop) :=
+quotient.induction_on s (by simpa using list.trop_sum)
+
+lemma trop_sum [add_comm_monoid R] (s : finset S) (f : S → R) :
+  trop (∑ i in s, f i) = ∏ i in s, trop (f i) :=
+begin
+  cases s,
+  convert multiset.trop_sum _,
+  simp
+end
+
+lemma fintype.trop_sum [add_comm_monoid R] [fintype S] (f : S → R) :
+  trop (∑ i : S, f i) = ∏ i : S, trop (f i) :=
+trop_sum _ _
+
+lemma list.untrop_prod [add_monoid R] (l : list (tropical R)) :
+  untrop l.prod = list.sum (l.map untrop) :=
+begin
+  induction l with hd tl IH,
+  { simp },
+  { simp [←IH] }
+end
+
+lemma multiset.untrop_prod [add_comm_monoid R] (s : multiset (tropical R)) :
+  untrop s.prod = multiset.sum (s.map untrop) :=
+quotient.induction_on s (by simpa using list.untrop_prod)
+
+lemma untrop_prod [add_comm_monoid R] (s : finset S) (f : S → tropical R) :
+  untrop (∏ i in s, f i) = ∑ i in s, untrop (f i) :=
+begin
+  cases s,
+  convert multiset.untrop_prod _,
+  simp
+end
+
+lemma fintype.untrop_prod [add_comm_monoid R] [fintype S] (f : S → tropical R) :
+  untrop (∏ i : S, f i) = ∑ i : S, untrop (f i) :=
+untrop_prod _ _
+
+lemma list.trop_minimum [linear_order R] (l : list R) :
+  trop l.minimum = list.sum (l.map (trop ∘ coe)) :=
+begin
+  induction l with hd tl IH,
+  { simp },
+  { simp [list.minimum_cons, ←IH] }
+end
+
+lemma multiset.trop_inf [h : linear_order R] [h' : order_top R] (s : multiset R) :
+  trop s.inf = multiset.sum (s.map trop) :=
+begin
+  induction s using multiset.induction with s x IH,
+  { simp },
+  { simp [←IH] }
+end
+
+lemma finset.trop_inf [h : linear_order R] [h' : order_top R] (s : finset S) (f : S → R) :
+  trop (s.inf f) = ∑ i in s, trop (f i) :=
+begin
+  cases s,
+  convert multiset.trop_inf _,
+  simp
+end
+
+lemma trop_Inf_image [conditionally_complete_linear_order R] (s : finset S)
+  (f : S → with_top R) : trop (Inf (f '' s)) = ∑ i in s, trop (f i) :=
+begin
+  rcases s.eq_empty_or_nonempty with rfl|h,
+  { simp only [set.image_empty, finset.coe_empty, finset.sum_empty, with_top.cInf_empty, trop_top] },
+  rw [←finset.inf'_eq_cInf_image _ h, finset.inf'_eq_inf],
+  convert s.trop_inf f,
+  refine lattice.ext _,
+  intros,
+  exact iff.rfl
+end
+
+lemma trop_infi [conditionally_complete_linear_order R] [fintype S] (f : S → with_top R) :
+  trop (⨅ (i : S), f i) = ∑ (i : S), trop (f i) :=
```

Comment:
It would be good to have also:
```lean
lemma trop_binfi [conditionally_complete_linear_order R] (s : finset S) (f : S → with_top R) :
  trop (⨅ i \mem s, f i) = ∑ i in s, trop (f i) :=
```

Reply:
I think this would require a lot more API for `conditionally_complete_lattice` that we don't have yet.

Reply:
Oops, its not even true! Nested `infi` (such as `binfi`) is largely meaningless on conditionally_complete_lattices, sadly.